typescript/src/index.ts

```diff
@@ -0,0 +1,26 @@
+import 'cross-fetch/polyfill';
+import BitriseAPI from './client/api';
+import { AuthTokenInterceptor, CSRFTokenInterceptor, InterceptorChain } from './client/auth-interceptors';
+import { CookieTokenStorage, ConstantTokenStorage, TokenStorage } from './client/storage';
+
+const createStorage = (token?: string | null): TokenStorage => {
+    if (token) {
+        return new ConstantTokenStorage(token);
+    }
+
+    return new CookieTokenStorage({
+        CSRFTokenName: 'CSRF-TOKEN',
+        authTokenName: 'expiring_api_token'
+    });
+};
+
+export default (apiToken?: string | undefined): BitriseAPI => {
```

Comment:
we support 2 modes:
* auth token specified and then we try to use that throughout
* auth token not specified -> try to authenticate but this will only work from bitrise domain